docs/design/object-locking.md

```diff
@@ -0,0 +1,187 @@
+# Object Locking using S3 Object Lock
+
+Object locking feature is built for addressing use cases where
+Write Once Read Many (WORM) model is required. The feature will be built in line
+with AWS specification, any extensions to the specification will be explicitly
+documented. The feature implementation has the goal of meeting
+SEC 17a-4 compliance.
+
+## Requirements
+
+* Bucket needs to have object lock flag set during bucket creation
+* Versioning has to be enabled on the bucket
+* Lock configuration can be written only on a bucket using PUT Object Lock
```

Comment:
Object lock must be enabled on a bucket in order to write a lock configuration using the PUT .... API